test/test_synthetic.py

```diff
@@ -0,0 +1,407 @@
+    Obj = modelName2Module(model)[1]()
+    ds = xr.open_dataset(wm_file)
+    t  = ds['t']
+    p  = ds['p']
+    e  = ds['e']
+    if kind == 'hydro':
+        p = t
+    elif kind == 'wet1':
+        e = t
+        Obj._k3 = 0
+    elif kind == 'wet2':
+        e = t**2
+        Obj._k2 = 0
+
+    Obj._t  = t
+    Obj._p  = p
+    Obj._e  = e
+
+    # make new delays and overwrite weather model dataset
+    Obj._get_wet_refractivity()
+    Obj._get_hydro_refractivity()
+
+    ds['wet']   = Obj._wet_refractivity
+    ds['hydro']   = Obj._hydrostatic_refractivity
+
+    os.makedirs(wm_dir, exist_ok=True)
+    dst = op.join(wm_dir, op.basename(wm_file))
+
+    if op.exists(dst):
+        os.remove(dst)
+
+    ds.to_netcdf(dst)
+
+    ds.close()
+    del ds
+    print ('Wrote synthetic weather model file to:', dst)
+    return dst
+
+
+def length_of_ray(target_xyz:list, model_zs, los):
+    """ Build rays at xy locations
+
+    Target xyz is a list of lists (xpts, ypts, hgt_levels)
+    Model_zs are all the model levels over which ray is calculated
+    los in los object (has the orbit info)
+    """
+
+    # Create a regular 2D grid
+    xx, yy = np.meshgrid(target_xyz[0], target_xyz[1])
+    hgt_lvls = target_xyz[2]
+
+    # Where total rays are stored,
+    outputArrs = np.zeros((hgt_lvls.size, target_xyz[1].size, target_xyz[0].size))
+
+    # iterate over height levels
+    for hh, ht in enumerate(hgt_lvls):
+
+        llh = [xx, yy, np.full(yy.shape, ht)]
+
+        xyz = np.stack(lla2ecef(llh[1], llh[0], np.full(yy.shape, ht)), axis=-1)
+        LOS = los.getLookVectors(ht, llh, xyz, yy)
+
+        cos_factor = None
+
+        # 2d array where output is added to; one per hgt lvl
+        outSubs = outputArrs[hh, ...]
+        # iterate over all model levels
+        for zz in range(model_zs.size-1):
+            low_ht = model_zs[zz]
+            high_ht = model_zs[zz + 1]
+
+            if (high_ht <= ht) or (low_ht >= 50000):
+                continue
+
+            # If high_ht > max_tropo_height - integral only up to max tropo
+            if high_ht > 50000:
+                high_ht = 50000
+
+            # If low_ht < height of point - integral only up to height of point
+            if low_ht < ht:
+                low_ht = ht
+
+            # Continue only if needed - 1m troposphere does nothing
+            if np.abs(high_ht - low_ht) < 1.0:
+                continue
+
+            low_xyz = getTopOfAtmosphere(xyz, LOS, low_ht, factor=cos_factor)
+            high_xyz = getTopOfAtmosphere(xyz, LOS, high_ht, factor=cos_factor)
+
+            ray_length = np.linalg.norm(high_xyz - low_xyz, axis=-1)
+
+            outSubs   += ray_length
+
+            cos_factor = (high_ht - low_ht) / ray_length if cos_factor is None else cos_factor
+
+    return outputArrs
+
+
+@dataclass
+class StudyArea(object):
+    """ Object with shared parameters related to the study area """
+    reg:str
+    WM:str
```

Comment:
Don't you use WM for classes sometimes? Please make the variable also more descriptive. Not upper case.

Reply:
https://github.com/bbuzz31/RAiDER/blob/74535ed4baac1d960ec7f6f3431f9a202d16e6e0/test/__init__.py#L23